catchup/catchpointService.go

```diff
@@ -824,6 +824,8 @@ func (cs *CatchpointCatchupService) checkLedgerDownload() error {
 		if err == nil {
 			return nil
 		}
+		// a non-nil error means that the catchpoint is not available, so we should rank it accordingly
+		cs.blocksDownloadPeerSelector.rankPeer(psp, peerRankNoCatchpointForRound)
```

Comment:
I suppose this function penalizes by the given amount? It's not setting an absolute rank?

Reply:
The class based peer selector's rankPeer increments a counter for download failures (a +1), the rankBasedPeerSelector runs the existing algorithm for ranking. Neither I would describe as absolute rank.

Reply:
Maybe sense